lib/kubernetes-deploy/kubernetes_resource/redis.rb

```diff
@@ -41,6 +43,15 @@ def service_ready?
       @service.dig("spec", "clusterIP").present?
     end
 
+    def name
+      return unless @definition.present?
```

Comment:
doesn't really matter, but this guard is unnecessary. `@definition` is the loaded YAML from their deploy directory; if it weren't present (minimally having `kind: Redis`), no instance of this class would be created.